docs/code/microsoft-ml-tokenizers-migration-guide.md

```diff
@@ -0,0 +1,19 @@
+# Porting to Microsoft.ML.Tokenizers
+
+This guide provides general guidance on how to migrate from various tokenizer libraries to `Microsoft.ML.Tokenizers` for Tiktoken.
+
+## Microsoft.DeepDev.TokenizerLib
+
+### API Guidance
+
+| Microsoft.DeepDev.TokenizerLib | Microsoft.ML.Tokenizers
+| --- | --- |
+| [TikTokenizer](https://github.com/microsoft/Tokenizer/blob/2c9ba5d343de52eb27521afef7c0c2f0f76c9c52/Tokenizer_C%23/TokenizerLib/TikTokenizer.cs#L20) | [Tokenizer](https://github.com/dotnet/machinelearning/blob/acced974bea6ed484503a595d87a3e7016c8a558/src/Microsoft.ML.Tokenizers/Tokenizer.cs#L28) |
+| [ITokenizer](https://github.com/microsoft/Tokenizer/blob/2c9ba5d343de52eb27521afef7c0c2f0f76c9c52/Tokenizer_C%23/TokenizerLib/ITokenizer.cs#L7) | [Tokenizer](https://github.com/dotnet/machinelearning/blob/acced974bea6ed484503a595d87a3e7016c8a558/src/Microsoft.ML.Tokenizers/Tokenizer.cs#L28) |
+| [TokenizerBuilder](https://github.com/microsoft/Tokenizer/blob/2c9ba5d343de52eb27521afef7c0c2f0f76c9c52/Tokenizer_C%23/TokenizerLib/TokenizerBuilder.cs#L14) | [TikToken.CreateByModelNameAsync](https://github.com/dotnet/machinelearning/blob/acced974bea6ed484503a595d87a3e7016c8a558/src/Microsoft.ML.Tokenizers/Model/Tiktoken.cs#L778) downloads<br>[Tiktoken.CreateByModelName[Async](Stream)](https://github.com/dotnet/machinelearning/blob/acced974bea6ed484503a595d87a3e7016c8a558/src/Microsoft.ML.Tokenizers/Model/Tiktoken.cs#L106-L180) user provided file stream |
+
+### General Guidance
+
+- Avoid hard-coding tiktoken regexes and special tokens.  Instead use the appropriate Tiktoken.`CreateByModelNameAsync` method to create the tokenizer from either a downloaded file, or a provided stream.
+- Avoid doing encoding if all you need is token count.  Instead use `Tokenizer.CountTokens`.

```

Comment:
>Avoid doing encoding if all you need is token count.  Instead use `Tokenizer.CountTokens`. [](http://example.com/codeflow?start=2&length=90)

Avoid doing encoding if you need the token count or encoded Ids.  Instead use `Tokenizer.CountTokens`. for getting the token count and `Tokenizer.EncodeToIds` for getting the encoded ids.

Reply:
Fixed